araproc/analysis/standard_reco.py

```diff
@@ -924,3 +926,346 @@ def get_avg_corr_snr(self, wave_packet, excluded_channels = []):
+            #   about to be trimmed. Switch to trimming off the front entirely.
+            front_trim = trim
+            back_trim = 0
+
+        # Return the results
+        if back_trim == 0:
+            return times[front_trim:], values[front_trim:]
+        else: 
+            return times[front_trim:-back_trim], values[front_trim:-back_trim]
+
+    def get_csw(
+        self, data, useful_event, solution, polarization, reco_results,
+        excluded_channels=[], which_distance='distant', return_csw=False
+    ):
+        """
+        Build the Coherently Summed Waveform (CSW) for the given `useful_event`
+          by shifting each waveform accoriding to the time delay calculated 
+          after finding the max cross correlation between two channels 
+          in a window around the expected arrival delay. 
+
+        Parameters
+        ----------
+        data : AnalysisDataset
+        useful_event : UsefulAtriStationEvent
+        solution : int
+            0 for direct ray solution. 1 for reflected/refracted ray solution.
+        polarization : int
+            0 for VPol, 1 for HPol
+        reco_results : dict
+            Reco results already with the specific reconstruction reqeusted (so
+            the keys of this object include 'theta' and 'phi')
+        excluded_channels : list
+            Any channels to exclude that aren't already excluded by 
+            livetime configurations or polarization. 
+        which_distance : str
+            The distance of the reconstruction for analysis, following the 
+            convention of other functions in this class. 
+        return_csw : bool
+            If True, will return the CSW_SNR and a ROOT.TGraph of the CSW
+
+        Returns
+        -------
+        csw : ROOT.TGraph
+            The coherently summed waveform. Only returned if `return_csw==True`
+        """
+
+        # Check that reco_results have been passed in properly
+        if 'theta' not in reco_results.keys():
+            raise ValueError(
+                "reco_result object passed into this function does not appear "
+                "to be for a specific reconstruction. Should have a 'theta' key "
+                "but only has the following:", reco_results.keys())
+        
+        # Load wavepacket with waveforms
+        wavepacket= data.get_wavepacket(
+            useful_event=useful_event, which_traces='filtered')
+
+        # Determine which channels should be used in the csw
+        if excluded_channels is not None: 
+            if not isinstance(excluded_channels, np.ndarray):
+                excluded_channels = np.array(excluded_channels)
+        else: 
+            excluded_channels = np.array([])
+        channels_to_csw = []
+        for ch_ID in data.rf_channel_indices: 
+            if ((ch_ID not in excluded_channels) and 
+                (data.rf_channel_polarizations[ch_ID] == polarization)):
```

Comment:
Is this check necessary? I wonder if we should retain the ability to form a csw between polarizations? (If we do drop this we should remove `polarization` as an argument)

Reply:
So, I think it's a design decision. And so far, we've decided to _exclude_ channels, not tell it what to include. So I'd rather keep that, if that's alright?

Reply:
Sure I can revert this commit and switch it to exclusion. The arrival delay functions also work via inclusion, would you like me to switch those to exclusion as well?

Reply:
I switched all of the functions I've added to work on the basis of exclusion of channels rather and inclusion.